gulpfile.js

```diff
@@ -12,13 +13,33 @@ var buildRoot = "_build";
 var packageRoot = "_package";
 var extnBuildRoot = "_build/Extensions/";
 var sourcePaths = "Extensions/**/*";
+var ExtensionFolder = "Extensions";
 
 gulp.task("clean", function() {
     return del([buildRoot, packageRoot]);
 });
 
 gulp.task("compile", ["clean"], function(done) {
-    return gulp.src(sourcePaths, { base: "." }).pipe(gulp.dest(buildRoot));
+    
+    if(args.testAreaPath === undefined )
+    {
+        return gulp.src(sourcePaths, { base: "." }).pipe(gulp.dest(buildRoot)); 
+    }
+    else
+    {     
+        var areaPathArgument = args.testAreaPath;
+        if(areaPathArgument.length > 0 )
+        {
+            console.log('Test area-paths - ' + areaPathArgument + ' are passed, compiling only specific area-path');        
```

Comment:
Make it more readable log something like "Building changed modules under area paths : ABC, XYZ"